libraries/lib-numeric-formats/NumericConverter.h

```diff
@@ -24,31 +24,30 @@
  * to the user */
 struct BuiltinFormatString;
 
-struct NumericField
+struct NUMERIC_FORMATS_API NumericField final
 {
 public:
-   NumericField(bool _frac, int _base, int _range, bool _zeropad)
-   {
-      frac = _frac;
-      base = _base;
-      range = _range;
-      zeropad = _zeropad;
-      digits = 0;
-   }
+   NumericField(bool _frac, int _base, int _range, bool _zeropad) noexcept;
+   
    NumericField( const NumericField & ) = default;
    NumericField &operator = ( const NumericField & ) = default;
    //NumericField( NumericField && ) = default;
    //NumericField &operator = ( NumericField && ) = default;
-   void CreateDigitFormatStr();
+   
    bool frac; // is it a fractional field
    int base;  // divide by this (multiply, after decimal point)
    int range; // then take modulo this
-   int digits;
-   int pos;   // Index of this field in the ValueString
-   bool zeropad;
+   int digits { 0 };
+      
    wxString label;
    wxString formatStr;
    wxString str;
+
+   int pos { -1 }; // Index of this field in the ValueString
```

Comment:
Seeing `-1` made me pause in my review to reason about all uses of `pos` to be sure there was no missing bounds checks.

I would instead write `size_t pos{ wxString::npos };` but seeing that, still I would do the same check.

Really the safety condition is this:  when `ParseFormatString` builds the vector of `NumericField`, it is always a legal subscript in `mValueTemplate`, which will also have the same length as `mValueString` when that is built.

That should be commented somewhere

Reply:
`mValueTemplate` was never read. It haven't survived :-)

Reply:
In any case - it was left uninitialized previously, so it really had some random value. I need to revisit the code, probably we can remove the `pos` as well

Reply:
In later code you will find the checks (but somewhat too late). This can be fixed easily